File: server-replica/nginx/test/test-cap.lua
-- max_seconds must end the response instead of streaming forever.
local p = require("probes")
local r = p.read_for("/_capped", 8)   -- endpoint is capped at 2s
if not r then ngx.say("ERROR") return end
ngx.say(string.format("elapsed=%.1f bytes=%d", r.elapsed, r.bytes))

File: server-replica/nginx/test/test-cap.js
// max_seconds must end the response instead of streaming forever.
import probes from 'probes.js';

async function run(r) {
    try {
        let result = await probes.readAll('/_capped');   // endpoint is capped at 2s
        r.return(200, 'elapsed=' + result.elapsed.toFixed(1) +
            ' bytes=' + result.bytes + '\n');
    } catch (e) {
        r.return(200, 'ERROR: ' + e.message + '\n');
    }
}

export default { run };
